### ISSUE TITLE
net.sqlcipher.database.SQLiteDatabase breaks formal contract with SupportSQLiteDatabase

### ISSUE THREAD

**Issue by gtomassoni** [sqlcipher/android-database-sqlcipher#588]
In the [androidx.sqlite.db.SupportSQLiteDatabase API description](https://developer.android.com/reference/androidx/sqlite/db/SupportSQLiteDatabase), Google states that SupportSQLiteDatabase implementations should throw an `android.database.SQLException` at least in case of invalid sql command string and, more generally, in case of a db-related error.

See, in example, [execSQL](https://developer.android.com/reference/androidx/sqlite/db/SupportSQLiteDatabase#execSQL(java.lang.String,%20java.lang.Object[])and [android.database.SQLException](https://developer.android.com/reference/android/database/SQLException.html).

`net.sqlcipher.database.SQLiteDatabase` throws instead `net.sqlcipher.database.net.sqlcipher.SQLException`.

This may have impacts in the smooth transition from the android's "stock" SupportSQLiteDatabase implementation to SQLChiper one.

### Expected Behavior

execSQL("Wrong Statement") throws a `android.database.SQLException`-based exception.

### Actual Behavior

execSQL("Wrong Statement") throws `net.sqlcipher.database.net.sqlcipher.SQLException`, which is not based on `net.sqlcipher.database.SQLiteDatabase`.

### Steps to Reproduce

SQLCipher for Android version: 4.5.1

**Comment by developernotes** [comment]
Hi @gtomassoni 

We are considering a change to have `net.sqlcipher.SQLException` extend `android.database.SQLException` for a future release of the software. This should help address this behavior.

**Comment by colinmacdonald** [comment]
I believe this issue is preventing Android Room's new `Upsert` feature from working when using sqlcipher: https://developer.android.com/jetpack/androidx/releases/room.

They are checking what the exception is to differentiate between a foreign/unique key [here](https://android-review.googlesource.com/c/platform/frameworks/support/+/2191947/9/room/room-runtime/src/main/java/androidx/room/EntityUpsertionAdapter.kt) but expect it to be `android.database.sqlite.SQLiteConstraintException` not a `net.sqlcipher.database.SQLException`

**Comment by techartist** [comment]
Yes, this issue is happening in my production code and a fix would be nice.

**Comment by mpost** [comment]
Just stumbled over the mismatching exceptions as well. Would be great to see fixed as we know need to rewrite an `@Upsert` manually.

**Comment by AlexTrotsenko** [comment]
@developernotes  any updates? Since it's blocking new `Upsert` in Room - this issue become quite problematic.

**Comment by developernotes** [comment]
Hi @AlexTrotsenko,

We will include `@Upsert` support in the next release of SQLCipher for Android.

**Comment by mpost** [comment]
@developernotes Of course there has to be this obligatory question of the ETA?

**Comment by developernotes** [comment]
Hi @mpost,

We don't have date to share for the next release at this time. You can subscribe to the release announcements here [^1] to be notified when the release becomes publicly available.

[^1]: https://discuss.zetetic.net/c/sqlcipher/updates/14

**Comment by mpost** [comment]
Ok, thanks for answering so quickly. 👍

**Comment by developernotes** [comment]
Hi folks,

`@Upsert` support has been added to this SQLCipher for Android repository, binaries will become available with our next public release, however, if you're able to build from source you can try it yourself now. 

Please note that with this series of changes we removed many of the parallel `net.sqlcipher.database` exception classes, favoring the usage of the `android.database.sqlite` variants due to the usage in `androidx`. When using the new builds, you will need to adjust your client code accordingly, often this should be limited to the `import` statements used:

```
-import net.sqlcipher.database.SQLiteException;
+import android.database.sqlite.SQLiteException;
```

**Comment by ItzNotABug** [comment]
Any updates on this fix release?
